packages/cosmic-swingset/src/anylogger-agoric.js

```diff
@@ -4,68 +4,66 @@ import {
+import { defineName } from '@agoric/internal/src/js-utils.js';
 
-// Turn on debugging output with DEBUG=agoric
+/** @import {BaseLevels} from 'anylogger'; */
+/** @typedef {keyof BaseLevels} LogLevel; */
 
 const DEBUG_LIST = getEnvironmentOptionsList('DEBUG');
 
-const isVatLogNameColon = nameColon =>
-  ['SwingSet:ls:', 'SwingSet:vat:'].some(sel => nameColon.startsWith(sel));
-
-// Turn on debugging output with DEBUG=agoric or DEBUG=agoric:${level}
-let selectedLevel =
-  DEBUG_LIST.length || getEnvironmentOption('DEBUG', 'unset') === 'unset'
+/**
+ * As documented in ../../../docs/env.md, the log level defaults to "log" when
+ * environment variable DEBUG is non-empty or unset, and to the quieter "info"
+ * when it is set to an empty string, but in either case is overridden if DEBUG
+ * is a comma-separated list that contains "agoric:none" or "agoric:${level}" or
+ * "agoric" (the last an alias for "agoric:debug").
+ *
+ * @type {string | undefined}
+ */
+let maxActiveLevel =
+  DEBUG_LIST.length > 0 || getEnvironmentOption('DEBUG', 'unset') === 'unset'
     ? 'log'
     : 'info';
 for (const selector of DEBUG_LIST) {
-  const parts = selector.split(':');
-  if (parts[0] !== 'agoric') {
-    continue;
-  }
-  if (parts.length > 1) {
-    selectedLevel = parts[1];
-  } else {
-    selectedLevel = 'debug';
+  const fullSelector = selector === 'agoric' ? 'agoric:debug' : selector;
+  const [_, detail] = fullSelector.match(/^agoric:(.*)/gs) || [];
+  if (detail) {
+    maxActiveLevel = detail === 'none' ? undefined : detail;
   }
 }
-const selectedCode = anylogger.levels[selectedLevel];
-const globalCode = selectedCode === undefined ? -Infinity : selectedCode;
+const maxActiveLevelCode = /** @type {number} */ (
+  (maxActiveLevel && anylogger.levels[maxActiveLevel]) ?? -Infinity
+);
 
 const oldExt = anylogger.ext;
 anylogger.ext = logger => {
-  const l = oldExt(logger);
-  l.enabledFor = lvl => globalCode >= anylogger.levels[lvl];
+  logger = oldExt(logger);
 
-  const prefix = l.name.replace(/:/g, ': ');
+  /** @type {(level: LogLevel) => boolean} */
+  const enabledFor = level => anylogger.levels[level] <= maxActiveLevelCode;
+  logger.enabledFor = enabledFor;
 
-  const nameColon = `${l.name}:`;
-  const logBelongsToVat = isVatLogNameColon(nameColon);
+  const nameColon = `${logger.name}:`;
+  const label = logger.name.replaceAll(':', ': ');
 
-  // If this is a vat log, then it is enabled by a selector in DEBUG_LIST.
-  const logMatchesSelector =
-    !logBelongsToVat ||
-    DEBUG_LIST.some(selector => {
-      const selectorColon = `${selector}:`;
-      return nameColon.startsWith(selectorColon);
-    });
+  // Vat logs are suppressed unless matched by a prefix in DEBUG_LIST.
+  const isForVat =
+    nameColon.startsWith('SwingSet:ls:') || // "ls" for "liveslots"
+    nameColon.startsWith('SwingSet:vat:');
+  const suppressed =
+    isForVat && !DEBUG_LIST.some(prefix => nameColon.startsWith(`${prefix}:`));
```

Comment:
I would prefer if instead of burying the constant strings used in this logic (which is mainly log-level name math), they would continue to be surfaced in a module-level function (with a name and implementation of your choosing):

```js
// module-level function (season to taste...)
const isForVat = nameColon => ['SwingSet:ls:', 'SwingSet:vat:'].some(prefixColon => nameColon.startsWith(prefixColon));

// ...

anylogger.ext = logger => {

  // ...

  // Vat logs are suppressed unless matched by a prefix in DEBUG_LIST.
  const suppressed =
    isForVat(nameColon) && !DEBUG_LIST.some(prefix => nameColon.startsWith(`${prefix}:`));
```

Reply:
Exposed as a module-level `VAT_LOGGER_PREFIXES` array containing "SwingSet:vat" and "SwingSet:ls".